src/result.test.ts

```diff
@@ -15,104 +15,112 @@ const mockErr: Result<string, number> = err('message');
 const onOk = (a: number): number => a + 2;
 const onError = (e: string): string => `Output: ${e}`;
 
-
 // ----- Tests ----- //
 
 describe('either', () => {
-    it('runs the correct function when Result is Ok', () => {
-        expect(either<string, number, number | string>(onError, onOk)(mockOk)).toBe(6);
-    });
-
-    it('runs the correct function when Result is Err', () => {
-        expect(either<string, number, number | string>(onError, onOk)(mockErr))
-            .toBe('Output: message');
-    });
+	it('runs the correct function when Result is Ok', () => {
+		expect(
+			either<string, number, number | string>(onError, onOk)(mockOk),
+		).toBe(6);
+	});
+
+	it('runs the correct function when Result is Err', () => {
+		expect(
+			either<string, number, number | string>(onError, onOk)(mockErr),
+		).toBe('Output: message');
+	});
 });
 
 describe('map', () => {
-    it('runs the function when Result is Ok', () => {
-        expect(pipe2(
-            mockOk,
-            map(onOk),
-            either(identity, identity),
-        )).toBe(6);
-    });
-
-    it('passes the error through when Result is Err', () => {
-        expect(pipe2(
-            mockErr,
-            map(a => `Output: ${a}`),
-            either(identity, identity),
-        )).toBe('message');
-    });
+	it('runs the function when Result is Ok', () => {
+		expect(pipe2(mockOk, map(onOk), either(identity, identity))).toBe(6);
+	});
+
+	it('passes the error through when Result is Err', () => {
+		expect(
+			pipe2(
+				mockErr,
+				map((a) => `Output: ${a}`),
```

Comment:
I think this should fix it - looks like TS was helpfully narrowing the type of `mockErr`, but threw away the type information that told it the type of `a` 😬 

```suggestion
				map((a: number) => `Output: ${a}`),
```